src/tool_provider/artifactory.rs

```diff
@@ -18,19 +26,120 @@ impl ArtifactoryProvider {
         Self { paths }
     }
 }
-#[allow(unused)]
+
 impl ToolProviderImpl for ArtifactoryProvider {
-    fn get_releases(&self, repo: &str) -> ForemanResult<Vec<Release>> {
-        Err(ForemanError::Other {
-            message: "Artifactory is not yet supported. Please use Github or Gitlab as your source"
-                .to_owned(),
-        })
+    fn get_releases(&self, repo: &str, host: &Url) -> ForemanResult<Vec<Release>> {
+        let client = Client::new();
+
+        let url = format!("{}artifactory/api/storage/{}", host, repo);
```

Comment:
Does `host` always automatically have a trailing `/`? Or must it be provided somewhere?

Reply:
Speaking of, could we potentially use `Url` methods to do this in a slightly more structured way? https://docs.rs/url/latest/url/struct.Url.html#method.join

Or would it not provide any meaningful benefit?

Reply:
host does automatically have the trailing `/`, but using something like url join is probably worth it.